crates/goose-server/src/state.rs

```diff
@@ -1,14 +1,14 @@
 use goose::agents::Agent;
 use goose::scheduler_trait::SchedulerTrait;
-use std::sync::atomic::AtomicUsize;
+use std::sync::atomic::{AtomicUsize, Ordering};
 use std::sync::Arc;
 use tokio::sync::Mutex;
 
 pub type AgentRef = Arc<Agent>;
 
 #[derive(Clone)]
 pub struct AppState {
-    agent: Option<AgentRef>,
+    agent: Arc<Mutex<Option<AgentRef>>>,
```

Comment:
this will now be an `Arc<Mutex<Option<Arc<Agent>>>>`

better to change the AgentRef type above? my cmd+f tells me it need not be `pub` either

Reply:
Maybe? I removed the optional which seems like a win. I am no Rust expert but to me this reads like, we have things that need an AgentRef, but we manage that AgentRef in a mutable fashion by sticking it into an Arc<Mutex

Reply:
`Arc<RwLock<Agent>>` is I think what you want